.github/workflows/frontend.yml

```
@@ -0,0 +1,38 @@
+name: frontend
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+  CI: true
+
+jobs:
+  test:
+    name: UI Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18.x
+      - name: Verify package-lock.json
+        working-directory: frontend
+        run: ./scripts/verify_lock.mjs
+      - name: Install
+        working-directory: frontend
+        run: npm clean-install --ignore-scripts
+      - name: Lint sources
+        working-directory: frontend
+        run: npm run lint
+      - name: Build
+        working-directory: frontend
+        run: npm run build
+      - name: Test
+        working-directory: frontend
+        run: npm run test -- --coverage --watchAll=false
```

Maybe use a default `working-directory` for all the steps?

```suggestion
    defaults:
      run:
        working-directory: backend

    steps:
      - uses: actions/checkout@v4
      - name: Use Node.js
        uses: actions/setup-node@v4
        with:
          node-version: 18.x
      - name: Verify package-lock.json
        run: ./scripts/verify_lock.mjs
      - name: Install
        run: npm clean-install --ignore-scripts
      - name: Lint sources
        run: npm run lint
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test -- --coverage --watchAll=false
```

---

Thanks I didn't know about this trick!